src/subdataframe/subdataframe.jl

```diff
@@ -112,12 +112,20 @@ index(sdf::SubDataFrame) = index(sdf.parent)
 nrow(sdf::SubDataFrame) = ncol(sdf) > 0 ? length(sdf.rows)::Int : 0
 ncol(sdf::SubDataFrame) = length(index(sdf))
 
+function Base.getindex(sdf::SubDataFrame, colind::Union{Symbol, Integer, Colon})
+    return sdf.parent[sdf.rows, colind]
+end
+
 function Base.getindex(sdf::SubDataFrame, colinds::Any)
-    return sdf.parent[sdf.rows, colinds]
+    return SubDataFrame(sdf.parent[colinds], sdf.rows)
+end
+
+function Base.getindex(sdf::SubDataFrame, rowinds::Any, colind::Union{Symbol, Integer, Colon})
+    return sdf.parent[sdf.rows[rowinds], colind]
 end
 
 function Base.getindex(sdf::SubDataFrame, rowinds::Any, colinds::Any)
-    return sdf.parent[sdf.rows[rowinds], colinds]
+    return SubDataFrame(sdf.parent[colinds], sdf.rows[rowinds])
```

Comment:
True, though I see it as being analogous to `v = view(d, d[:x] .== 1); view(v, v[:y] .== 2)`, which retains the parent, and composes the row indices.

I'm about to push an update which will include:
```julia
function Base.getindex(sdf::SubDataFrame, ::Colon, colinds::Any)
    return sdf.parent[sdf.rows, colinds]
end
```
to ensure that if `rowind === Colon()` a view is *not* returned.